### ISSUE TITLE
[Enhancement] Allow Find in File input to be cleared with "Esc" key or other keyboard shortcut

### ISSUE THREAD

**Issue by TraceyC77** [Zettlr/Zettlr#2970]
### Description

I use "Find in File" very frequently. I would like there to be an easier way, preferably with a keyboard shortcut, to clear the input

**Current behavior**
* Open Find in File and type in a search term
* Close the Find input
* Open Find in File again, the previous search term is displayed. (This is fine, I may want to search on that term again)
* The only way to clear the entire search term from the input is to use Backspace or select the entire term and use Delete. This is more cumbersome than a keyboard shortcut since it forces the use of the mouse. (If the search term has a space, Ctrl-Backspace only deletes the word after the space)

I have searched through previous reported issues and the documentation, and didn't see that there was any other way to clear search terms.

### Proposed Changes

Allow the "Esc" key (or another keyboard shortcut) to clear the Find in File search term when that input dialog is displayed.

Alternatively, have the search term highlighted when pressing Ctrl-F (to either bring up the Find in File dialog or re-focus it)
Change Ctrl-F to re-focus the Find in File dialog if it is already open (this is what web browsers do)

**Proposed behavior:**
* Open Find in File and type in a search term
* Close the Find input
* Open Find in File again, the previous search term is displayed. (This is fine, I may want to search on that term again)
* Press "Esc" to clear the search term (or the designated keyboard shortcut)
**Alternate proposal**
* Open the find in file dialog (or re-focus it) with Ctrl/Cmd-F and the search term highlighted.
* Pressing Delete will clear the search input.

### Caveats

I didn't see any impact of changing "Esc" to map to clearing the search terms. Currently, it will close the Find in File dialog.

Changing Ctrl-F to re-focus the dialog if it's already open may impact people who have gotten used to the current behavior. This might be a setting that can be added if this option is chosen.

### Do you Wish to Attempt Implementing this Yourself?

No

### Zettlr Version

Stable (most recent version)

### Your Platform

- [X] Windows
- [ ] macOS
- [X] Linux

### Operating System Version

Linux Solus

**Comment by Redsandro** [comment]
Agreed. I reckon the behavior should be similar to the _"Restrict search to directory"_ input where pressing `esc` clears the value.

**Comment by Redsandro** [comment]
This works for me and is already active (not sure since when).

Although technically I'd prefer consistency. In both Filter, Full Text Search and Restrict, let's call them "others", the behavior is different:

Input | Find dialog | All others
---|---|---
Shortcut key | Toggle | Show only, never hide
`Esc` | Hide | Clear
`Backspace` | Clear selection | Remove last character

Perhaps Ctrl+Shift+F can also work as a toggle, like Ctrl+F, and the word should also be selected, like with Ctrl+F.

**Comment by TraceyC77** [comment]
> Would having it select (so that you just need to press backspace once to clear the input) also work? This way we could reserve Escape for closing the search

This behavior was implemented in in-file search a couple of versions ago by another pull request. The way it works right now is nice because it's consistent with other editors and web browsers.

I agree with @Redsandro that consistency would be preferable. I don't use Full Text Search often, but it would be better if it worked the way File Find does right now.
